Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/DSDeferredResponseAuditTransforms.java

```diff
@@ -0,0 +1,231 @@
+import gov.hhs.fha.nhinc.docsubmission.audit.DocSubmissionAuditTransformsConstants;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import gov.hhs.fha.nhinc.properties.PropertyAccessException;
+import gov.hhs.fha.nhinc.properties.PropertyAccessor;
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformHelper;
+import gov.hhs.healthit.nhin.XDRAcknowledgementType;
+import java.lang.management.ManagementFactory;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Properties;
+import oasis.names.tc.ebxml_regrep.xsd.rs._3.RegistryResponseType;
+
+/**
+ *
+ * @author achidamb
+ */
+public class DSDeferredResponseAuditTransforms extends
+    AbstractDocSubmissionAuditTransforms<RegistryResponseType, XDRAcknowledgementType> {
+
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForRequest(RegistryResponseType request,
+        AssertionType assertion, AuditMessageType auditMsg) {
+        return auditMsg;
+    }
+
+    @Override
+    protected AuditMessageType getParticipantObjectIdentificationForResponse(RegistryResponseType request,
+        XDRAcknowledgementType response, AssertionType assertion, AuditMessageType auditMsg) {
+        return auditMsg;
+    }
+
+    @Override
+    protected String getServiceEventIdCodeRequestor() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_CODE_DS_RECIPIENT;
+    }
+
+    @Override
+    protected String getServiceEventIdCodeResponder() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_CODE_DS_SOURCE;
+    }
+
+    @Override
+    protected String getServiceEventCodeSystem() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_CODE_SYSTEM;
+    }
+
+    @Override
+    protected String getServiceEventDisplayRequestor() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_DISPLAY_RECIPIENT;
+    }
+
+    @Override
+    protected String getServiceEventDisplayResponder() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ID_DISPLAY_SOURCE;
+    }
+
+    @Override
+    protected String getServiceEventTypeCode() {
+        return DocSubmissionAuditTransformsConstants.EVENT_TYPE_CODE;
+    }
+
+    @Override
+    protected String getServiceEventTypeCodeSystem() {
+        return DocSubmissionAuditTransformsConstants.EVENT_TYPE_CODE_SYSTEM;
+    }
+
+    @Override
+    protected String getServiceEventTypeCodeDisplayName() {
+        return DocSubmissionAuditTransformsConstants.EVENT_TYPE_CODE_DISPLAY_NAME;
+    }
+
+    @Override
+    protected String getServiceEventActionCodeRequestor() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ACTION_CODE_RECIPIENT;
+    }
+
+    @Override
+    protected String getServiceEventActionCodeResponder() {
+        return DocSubmissionAuditTransformsConstants.EVENT_ACTION_CODE_SOURCE;
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        return null;
```

Comment:
yes. The HumanRequestor ActiveParticipant will be null for both Response and ack auditing. HumanRequestor activeparticipant is usulaly obtain from assertion obtain from incoming request. The request is not available while we audit response in case of Deferred.